internal/clients/api_client.go

```diff
@@ -246,8 +257,8 @@ func (a *ApiClient) GetFleetClient() (*fleet.Client, error) {
 	return a.fleet, nil
 }
 
-func (a *ApiClient) SetAlertingAuthContext(ctx context.Context) context.Context {
-	return context.WithValue(ctx, alerting.ContextBasicAuth, alerting.BasicAuth{
+func (a *ApiClient) SetGeneratedClientAuthContext(ctx context.Context) context.Context {
+	return context.WithValue(ctx, slo.ContextBasicAuth, slo.BasicAuth{
```

Comment:
Hmmm, good question. I see why this function used to have cursing in it. The acceptance tests passed, but perhaps this isn't used there?

Reply:
Do we really need to recreate this for each generated client?

Reply:
Nope, I broke it. 😂 
Annoyingly, this is exactly the same on both generated clients. Wondering how we can just have one version of this.

Reply:
Basically, we can't. We could do something like:

```
func SetAlertingAuthContext(ctx) ctx {
  return SetGeneratedClisnterAuthContext(ctx, alerting.ContextBasicAuth)
}

func SetGeneratedClisnterAuthContext(ctx, key any) ctx {
  return context.WithValue(ctx, key, ...)
}
```

But we can't reduce this to a single function. Golang context keys require strict type equivalence, `slo.contextKey("hello") != alerting.contextKey("hello")` since the underlying types are different even though they both derive from string. This is designed to avoid key collisions on common string values.

Reply:
TIL, good to know, will add a version for each generated client then.